x/wasm/types/types.go

```diff
@@ -83,6 +90,60 @@ func (c *ContractInfo) ValidateBasic() error {
 	if err := validateLabel(c.Label); err != nil {
 		return sdkerrors.Wrap(err, "label")
 	}
+	if c.Extension == nil {
+		return nil
+	}
+	e, ok := c.Extension.GetCachedValue().(validatable)
```

Comment:
This should definitely be documented (maybe below?)
not just how to register an Any type for the extension, but all variants of this type must implement `ValidateBasic()`

Reply:
uh, that was not intended. I had the interface implementation mandatory but made it optional later. Good to have you covering me. 😅